### PR TITLE
Add TLS support to the Trillian log signer

### PR DESCRIPTION
**Pull request by maljub01**
Addresses #428

### CODE REVIEW

File: server/trillian_log_signer/main.go [hunk 1]
```diff
@@ -48,6 +48,8 @@ import (
 var (
 	mySQLURI                 = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "Connection URI for MySQL database")
 	httpEndpoint             = flag.String("http_endpoint", "localhost:8091", "Endpoint for HTTP (host:port, empty means disabled)")
+	tlsCertFile              = flag.String("tls_cert_file", "", "Path to the TLS server certificate.")
```
Comment (AlCutter) [inline server/trillian_log_signer/main.go]: Perhaps add some text to say not configuring/setting and tlsKeyFile to empty results in a non-TLS server being run?

File: util/process.go [hunk 2]
```diff
@@ -25,14 +25,21 @@ import (
...
-func StartHTTPServer(addr string) error {
+func StartHTTPServer(addr, certFile, keyFile string) error {
 	sock, err := net.Listen("tcp", addr)
 	if err != nil {
 		return err
 	}
 	go func() {
 		glog.Info("HTTP server starting")
-		http.Serve(sock, nil)
+		if certFile != "" || keyFile != "" {
```
Comment (AlCutter) [inline util/process.go]: Can this work without both bring present, or are you just wanting to interpret either flag being non-empty as operator intent to run TLS but let `http.ServeTLS` 
 handle this potential error case? (maybe add a comment to that effect if that's what you're after)

Comment (maljub01) [reply util/process.go]: Yes, that was my intention. Added a comment.

File: server/trillian_log_signer/main.go (hunk 1 above)
Comment (maljub01) [reply server/trillian_log_signer/main.go]: Done.

File: server/trillian_log_signer/main.go [hunk 3]
```diff
@@ -48,8 +48,8 @@ import (
 var (
 	mySQLURI                 = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "Connection URI for MySQL database")
 	httpEndpoint             = flag.String("http_endpoint", "localhost:8091", "Endpoint for HTTP (host:port, empty means disabled)")
-	tlsCertFile              = flag.String("tls_cert_file", "", "Path to the TLS server certificate.")
-	tlsKeyFile               = flag.String("tls_key_file", "", "Path to the TLS server key.")
+	tlsCertFile              = flag.String("tls_cert_file", "", "Path to the TLS server certificate. If not set, the server will use unsecured connections.")
```
Comment (AlCutter) [inline server/trillian_log_signer/main.go]: nit: "not set" -> "unset"

Comment (maljub01) [reply server/trillian_log_signer/main.go]: Done

**Pull request merged**